gcovr/gcov.py

```diff
@@ -471,6 +471,29 @@ def run_with_args(self, args, **kwargs):
             **kwargs,
         )
 
+    def run_with_args(self, args, **kwargs):
+        """Run the gcov program
+
+        >>> gcov("bash").run_with_args(["-c", "exit 1"])
+        Traceback (most recent call last):
+        ...
+        RuntimeError: GCOV returncode was 1.
+        >>> gcov("bash").run_with_args(["-c", "kill $$"])
+        Traceback (most recent call last):
+        ...
+        RuntimeError: GCOV returncode was -15 (exited by signal).
```

Comment:
Hmm, seems like this is more complicated after all. The `gcov()` constructor invokes those programs and tries to determine valid command line options, which seems to cause test failures on Windows. I'm not sure how this can be sidestepped elegantly, in particular because the gcov class maintains significant global state.

My guess is that it would make sense to rename the gcov class (maybe `GcovProgram`?), with a constructor that doesn't do any sniffing. Then, a function `gcov()` could manage option checking + caching. This would avoid changes to other parts of the code.

Reply:
I'll try but instead of `gcov` I prefer something like `cache_gcov_capabilities()`.